content/en/docs/concepts/storage/persistent-volumes.md

```diff
@@ -785,6 +785,70 @@ spec:
       storage: 10Gi
 ```
 
+## Volume populators and data sources
+
+{{< feature-state for_k8s_version="v1.22" state="alpha" >}}
+
+{{< note >}}
+Kubernetes supports custom volume populators; this alpha feature was introduced
+in Kubernetes 1.18. Kubernetes 1.22 reimplemented the mechanism with a redesigned API.
+Check that you are reading the version of the Kubernetes documentation that matches your
+cluster. {{% version-check %}}
+To use custom volume populators, you must enable the `AnyVolumeDataSource`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) for
+the kube-apiserver and kube-controller-manager.
+{{< /note >}}
+
+Volume populators take advantage of a PVC spec field called `dataSourceRef`. Unlike the
+`dataSource` field, which can only contain either a reference to another PersistentVolumeClaim
+or to a VolumeSnapshot, the `dataSourceRef` field can contain a reference to any object in the
+same namespace, except for core objects other than PVCs. For clusters that have the feature
+gate enabled, use of the `dataSourceRef` is preferred over `dataSource`.
+
+## Data source references
+
+The `dataSourceRef` field behaves almost the same as the `dataSource` field. If either one is
+specified while the other is not, the API server will give both fields the same value. Neither
+field can be changed after creation, and attempting to specify different values for the two
+fields will result in a validation error. Therefore the two fields will always have the same
+contents.
+
+There are two differences between the `dataSourceRef` field and the `dataSource` field that
+users should be aware of:
+* The `dataSource` field ignores invalid values (as if the field was blank) while the
+  `dataSourceRef` field never ignores values and will cause an error if an invalid value is
+  used. Invalid values are any core object (objects with no apiGroup) except for PVCs.
```

Comment:
Is this correct:
```suggestion
   For the `dataSource` field, you can only set the `apiGroup` field to `""` (the empty string,
   representing the core {{< glossary_tooltip text="API Group" term_id="api-group" >}})
   if you specify its `kind` as `PersistentVolumeClaim`; if you specify a different `kind` and
   set `apiGroup` to `""` then the API server rejects the request.
   For the `dataSourceRef` field you can specify any `kind` and `apiGroup`; however, the
   only kind within the core API group that can act as a data source is still PersistentVolumeClaim.
   If you try to create a PersistentVolumeClaim using some other core API as a data source,
   the PersistentVolumeClaim you create is invalid.
```
? I couldn't be sure I understood the text I've tried to reword.